### PR TITLE
Backport of ScaleProfile from ImaGene

### PR DESCRIPTION
**Pull request by kerautret**
Todo:
- [X] backport main imagene class
- [X] add catch test
- [x] check DGtal naming 
- [x] complete doc

### CODE REVIEW

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  {
+    // ----------------------- Standard services ------------------------------
+  public:
+
+    /**
+     * Used to specify the method to compute the profile values
+     * (used in @ref getProfile())
+     * 
+     **/
+    enum ProfileComputingType{MEAN, MAX, MIN, MEDIAN};
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: maybe ScaleProfileType or ScaleProfileCategory

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+    /**
+     * Initializer. Must be called before adding datas. Specifies the
+     * scales of the profile (generally an iterator on a sequence
+     * (1,2,3,4,...N)).
+     *
+     * @param[in] beginScale an iterator pointing on the first scale (some
+     * floating-point convertible value).
+     * @param[in] endScale an iterator pointing after the last scale.
+     * @param[in] storeValsInStats flat to store values in statistics (in order to be
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: (so that the median value is accessible)

Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: flat => flag

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+                const bool storeValsInStats=false );
+
+
+
+    /**
+     * Initializer. Must be called before adding datas. Specifies the
+     * scales of the profile as the sequence (1,2,3,4,...,nb).
+     *
+     * @param[in] nb an integer number strictly positive.
+     * @param[in] storeValsInStats flat to store values in statistics (in order to be
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: idem as above

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+     * @param stat any statistic (which is added to the current
+     * statistic object).
+     */
+    void addStatistic( unsigned int idxScale, const Statistic<float> & stat );
+
+
+
+    /**
+     * It stops and Erase the stats saved values. It must be called to
+     * avoid to store all statistics values when we have to acess to
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: acess => access

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+     * @code
+     * ScaleProfile sp;
+     * // the value are now stored and you can access to the median value of the profile.
+     * sp.init(true);
+     * sp.addValue(0, 10.5);
+     * sp.addValue(0, 9.2);
+     *  ...
+     * // When all values have been added you can stop to store them again
+     * sp.stopStatsSaving();
+     * // before to erase the statistics sample values the median is computed and stored.
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: before erasing all statistics data the median ...

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+    // ----------------------- Profile services --------------------------------
+  public:
+    
+    /**
+     * Used to define the method to determine the profile values from
+     * the samples of scale statistics.
+     *
+     * @param type the method applied to the statistics samples: MEAN, MAX, MIN.
+     **/    
+    void setProfileDef(ProfileComputingType type);
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: Perhape setType (if ScaleProfileType) or setCategory  (if ScaleProfileCategory).

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 7]
```diff
@@ -0,0 +1,406 @@
...
+		      const unsigned int minSize = 1,
+		      const double maxSlope = -0.2,
+		      const double minSlope = -1e10 ) const;
+
+    /**
+     *  Compute the profile slope of the first meaningful scale
+     *  interval computed by a simple linear regression model.
+     *
+     * @return a pair<bool, double> giving the slope and indicating if
+     * the a meaningful scale was well found or not. If no meaningful
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: the a => a

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 8]
```diff
@@ -0,0 +1,406 @@
...
+    
+
+
+    /**
+     * Used to temporaly store values in statistics in order to be
+     * able to access to the median value. By default the value is set
+     * to false and the median is not available. 
+     * @see setStoreStats
+     */
+    bool m_storeValInStats;
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: myStoreValInStats

File: src/DGtal/geometry/helpers/ScaleProfile.ih [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file ScaleProfile.ih
+ * @author Bertrand Kerautret (\c kerautre@loria.fr )
+ * LORIA (CNRS, UMR 7503), University of Nancy, France
+ *
+ * @Jacques-Olivier Lachaud
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/helpers/ScaleProfile.ih]: @author

File: tests/geometry/testScaleProfile.cpp [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+      sp.addValue(0, 10.0);
+      sp.addValue(1, 10.0);
+      sp.addValue(2, 10.0);
+      sp.addValue(3, 10.0);
+      sp.addValue(4, 10.0);
+      std::vector<double> x;
+      std::vector<double> y;
+      sp.getProfile(x, y);
+
+      REQUIRE( (x[3] == log(4)) );
```
Comment (JacquesOlivierLachaud) [inline tests/geometry/testScaleProfile.cpp]: Is there an approximate comparison in catch ?

File: tests/geometry/testScaleProfile.cpp [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      sp.init(scales.begin(), scales.end());  
+      sp.addValue(0,22);
+      sp.addValue(1,15);
+      sp.addValue(2,8);
+      sp.addValue(3,17);
+      sp.addValue(4,7);
+      sp.addValue(5,2);      
+      std::vector<double> x,y;
+      sp.getProfile(x,y); 
+      REQUIRE( (x[0] == log(5)) );
```
Comment (JacquesOlivierLachaud) [inline tests/geometry/testScaleProfile.cpp]: I think there is a catch for approximately equal.

File: src/DGtal/geometry/helpers/ScaleProfile.h (hunk 1 above)
Comment (kerautret) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: yes, changing with ScaleProfileType

File: src/DGtal/geometry/helpers/ScaleProfile.h (hunk 2 above)
Comment (kerautret) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: ok thanks

File: src/DGtal/geometry/helpers/ScaleProfile.h (hunk 6 above)
Comment (kerautret) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: yes so, setType ;)

File: tests/geometry/testScaleProfile.cpp (hunk 10 above)
Comment (kerautret) [inline tests/geometry/testScaleProfile.cpp]: Yes, David confirms me it !

File: tests/geometry/testScaleProfile.cpp (hunk 11 above)
Comment (kerautret) [inline tests/geometry/testScaleProfile.cpp]: yes it is "Approx()", thanks  I add it

**Comment by kerautret** [comment]
Thanks for the corrections, that done

File: ChangeLog.md [hunk 12]
```diff
@@ -7,6 +7,14 @@
    the width (vertical/horizontal or Euclidean) of a convex hull.
    (Bertrand Kerautret, [#1052](https://github.com/DGtal-team/DGtal/pull/1052))
 
+ - ScaleProfile: new class to represent a (multi)scale profile i.e. a sequence
+   of statistics on digital lengths parameterized by a scale.
```
Comment (dcoeurjo) [inline ChangeLog.md]: I don't understand the "digital lengths".. anything could be stored in a ScaleProfile, right ?

BTW I have the feeling that the ScaleProfile name for this is confusing..

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 13]
```diff
@@ -0,0 +1,406 @@
...
+  /////////////////////////////////////////////////////////////////////////////
+  // class ScaleProfile
+  /**
+   * Description of class 'ScaleProfile' <p> \brief Aim: This class
+   * represents one (multi)scale profile, i.e. a sequence of statistics
+   * on digital lengths parameterized by a scale. This class only
+   * represents what happens at one place, not everywhere on the
+   * contour. All further computations in the scale profile are
+   * done in logspace.
+   *
```
Comment (dcoeurjo) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: Please document the way this class should be used.

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 14]
```diff
@@ -0,0 +1,406 @@
...
+     * the samples of scale statistics.
+     *
+     * @param type the method applied to the statistics samples: MEAN, MAX, MIN.
+     **/    
+    void setType(ScaleProfileType type);
+    
+
+    
+    /**
+     * Used by boost to Serialize.
```
Comment (dcoeurjo) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: boost serialization is not used anymore in DGtal, shouldn't this be cleaned up ?

File: src/DGtal/geometry/helpers/ScaleProfile.h [hunk 15]
```diff
@@ -0,0 +1,406 @@
...
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // class ScaleProfile
+  /**
+   * Description of class 'ScaleProfile' <p> \brief Aim: This class
+   * represents one (multi)scale profile, i.e. a sequence of statistics
+   * on digital lengths parameterized by a scale. This class only
```
Comment (dcoeurjo) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: maybe be i.e. -> e.g. because it is more general than digital lengths analysis.

**Comment by dcoeurjo** [comment]
Ok It clearer what's puzzling me.. First, I know exactly why this class was developed  in imagene ;) 

But the "naming" is confusing: this class processes a sequence of statistics and ```meaningfullScale``` outputs the range of indices in the sequence satisfying a given statistical property (being linear model in logscale). The fact that indices of the sequence are "scales" is a restriction.

Furthermore, the fact that statistical properties are "log scaled" makes sense for the application but is not generic at all for DGtal.

Maybe renaming to "LogProfile" (to clearly indicate that log is performed)  and remove all mentions to scale ?

File: ChangeLog.md (hunk 12 above)
Comment (kerautret) [inline ChangeLog.md]: yes comment related to the meaningful scale ;)
I clean it

File: src/DGtal/geometry/helpers/ScaleProfile.h (hunk 13 above)
Comment (kerautret) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: ok

File: src/DGtal/geometry/helpers/ScaleProfile.h (hunk 14 above)
Comment (kerautret) [inline src/DGtal/geometry/helpers/ScaleProfile.h]: was done in ImaGene just for some entities for 3d manipulation it was really useful (without need to serialize all. But for now agree that it is not used so I remove and I will see if we need later.

**Comment by kerautret** [comment]
agree with you, LogProfile is fine but perhaps instead to remove scale ref, the use of LogScaleProfile ? Else we need to find other name for meaningfullScale or noiseLevel but if you have suggestion ?

**Comment by dcoeurjo** [comment]
If would say:
- ok for ```LogProfile```
- I would move Meaningful scale related things (meaningfulScale & noiseLevel) in another class dedicated to maximal DSS stuff as you did with @JacquesOlivierLachaud

**Comment by kerautret** [comment]
Do you mean in  SegmentComputerUtils ?
But the meaningfulScale and noise level can be more generic (potentially given in 3D with maximal plane).
Else we can move them in something like ScaleUtils in the same directory DGtal/geometry/helpers ?

**Comment by dcoeurjo** [comment]
I don't know...

```meaningfulScale``` and ```noiseLevel``` a very specific tools and quantities computed on a logarithmic profile of (x,y) pairs of data. The problem is more related to the method names: ```noiseLevel``` refers to what it is used for and not what the method does. 

I would suggest a ```MeaningfulScale``` struct or class with both methods and a description referring to the papers, what do you think ?

**Comment by JacquesOlivierLachaud** [comment]
You are right: meaningful scale, noise level, standard scale are specific interpretations of (awaited) linear data. Perhaps it could put in some DGtal::math or DGtal::math::stats along with LogProfile.
They could however be reused in Integral Invariants and 3D shape analysis, hence they deserve to be in DGtal (and not DGtalTools) in my opinion.

**Comment by dcoeurjo** [comment]
Yep I agree, they are very interesting and would be very useful in other part. That's the reason why we discuss so much to make them clearly visible in the library and properly named (not just methods of some LogProfile stat class).

**Comment by JacquesOlivierLachaud** [comment]
Ok. We have two options:
1) either limit changes, rename LogScaleProfile to LogProfile and have a MeaningfulScaleAnalysis (or LogScaleAnalysis) class where we put the different methods. I think they could be in some math/statistics part, instead of geometry helpers for me.
2) or create a true statistics package, with
  - statistics classes
  - Profile class (templated by a functor to get a Log-profile class)
  - LinearAnalysis class with regression
  - ScaleAnalysis with the methods above.

It is not the same amount of work.

**Comment by kerautret** [comment]
yes, perhaps we can start with the first solution, the renaming is done, I have just to change the scale related method in a new MeaningfulScaleAnalysis or other.
And the second solution can be done later by merging all into a Statistics package with the doc associated (as David suggest in the stat pr).

**Comment by kerautret** [comment]
but also ok to direct start the second solution and let custom LogProfile in a tools eventually...

**Comment by dcoeurjo** [comment]
Is this PR ready for review ?

**Comment by kerautret** [comment]
not yet I am on it perhaps sunday I hope ;)

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 16]
```diff
@@ -0,0 +1,299 @@
...
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // class MeaningfulScaleAnalysis
+  /**
+   * Description of class 'MeaningfulScaleAnalysis' <p> \brief Aim:
+   * This class proposes the implementation of different methods used
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: this class implements different...

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+   * to define the meaningful scale analysis as proposed in 
+   * \cite kerautret_meaningful_2012 . In particular, it exploits the
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: exploits -> uses

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 18]
```diff
@@ -0,0 +1,299 @@
...
+   * Profile class to represent a multi-scale profile and to compute a
+   * meaningful scale. It also permits to get a noise estimation from
+   * the given profile.
+   * 
+   *
+   * A typical use is to exploit from the length of maximal segments
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: A typical example is for instance the length of maximal segments obtained at different scales

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+   * obtained at different scales. We show here a simple example of
+   * use with a single profile:
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: (remove the last sentence?)

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 20]
```diff
@@ -0,0 +1,299 @@
...
+   *
+   *
+   * First we start to construct a Profile:
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: First, we construct a Profile

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+   *   sp.addValue(2,8);
+   *   sp.addValue(3,17);
+   *   sp.addValue(4,7);
+   *   sp.addValue(5,2);      
+   *
+   * @endcode
+   * Finally we can construct the MeaningfulScaleAnalysis object and obtain the meaningful scale:
+   * @code 
+   *  MeaningfulScaleAnalysis<Profile<LogFct>> msa(sp);
+   *  std::vector< std::pair<uint, uint> > interval;
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: interval -> intervals ?

File: src/DGtal/math/MeaningfulScaleAnalysis.h [hunk 22]
```diff
@@ -0,0 +1,299 @@
...
+   *   sp.addValue(4,7);
+   *   sp.addValue(5,2);      
+   *
+   * @endcode
+   * Finally we can construct the MeaningfulScaleAnalysis object and obtain the meaningful scale:
+   * @code 
+   *  MeaningfulScaleAnalysis<Profile<LogFct>> msa(sp);
+   *  std::vector< std::pair<uint, uint> > interval;      
+   *  msa.computeMeaningfulScales(interval, 1);
+   *  uint n = msa.noiseLevel();
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: uint -> unsigned int (not defined for all compilers)

File: src/DGtal/math/MeaningfulScaleAnalysis.ih [hunk 23]
```diff
@@ -0,0 +1,193 @@
...
+  std::pair<bool, double> resuSlope(false, 0.0);
+  std::vector< std::pair< unsigned int, unsigned int > > vectIntervals;
+  this->computeMeaningfulScales(vectIntervals, minSize, maxSlope, minSlope);
+  std::vector<typename TProfile::Value> vx;
+  std::vector<typename TProfile::Value> vy;
+
+  myProfile.getProfile( vx, vy );
+  // no meaningfulScale interval: 
+  
+  if(vectIntervals.size()!=0 && vectIntervals.at(0).first<vectIntervals.at(0).second ){
```
Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.ih]: { in new lines

Comment (dcoeurjo) [inline src/DGtal/math/MeaningfulScaleAnalysis.ih]: please check the indentation

File: src/DGtal/math/Profile.h [hunk 24]
```diff
@@ -0,0 +1,384 @@
...
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // class Profile
+  /**
+   * Description of class 'Profile' <p> \brief Aim: This class can be
+   * used to represent a profile (PX, PY) defined from an input set of
+   * samples (Xi, Yi). From all the samples (Xk, Yk) having the same
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: From -> For all sample...

File: src/DGtal/math/Profile.h [hunk 25]
```diff
@@ -0,0 +1,384 @@
...
+   * value Xk, the associated value PY is computed (by default) by the
+   * mean of the values Yk. Note that other definitions can be used
+   * (MAX, MIN or MEDIAN).
+   *
+   * 
+   * This class is templated by the type of the functor TValueFunctor
+   * which is applied on the two coordinate values of the resulting
+   * profile (PX, PY). This functor is by default set to the identity
+   * functor. The default type (float) of the values added to the
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: float -> double?

File: src/DGtal/math/Profile.h [hunk 26]
```diff
@@ -0,0 +1,384 @@
...
+   * profile can also be changed by using the template parameter
+   * TValue.
+   *
+   * For instance to construct a Profile defined from a data sample
+   * having its Xi integer values included from 1 to 10, you can construct
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: remove "included"?

File: src/DGtal/math/Profile.h [hunk 27]
```diff
@@ -0,0 +1,384 @@
...
+   * @code
+   * std::vector<float> xDef;
+   * for (float i = 0.5; i < 5; i=i+0.5){
+   *     xDef.push_back(i);
+   * }   
+   * Profile<> sp;
+   * sp.init(xDef.begin(), xDef.end());
+   * @endcode
+   *
+   * Then, you can add all the samples (Xk, Yk) :
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: remove "all"

Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: remove space before ":"

File: src/DGtal/math/Profile.h [hunk 28]
```diff
@@ -0,0 +1,384 @@
...
+   * @code
+   * sp.addValue(0, 23.0);
+   * sp.addValue(0, 20.0);
+   * sp.addValue(1, 20.2);
+   * sp.addValue(2, 10.4);
+   * ...
+   * sp.addValue(9, 20);
+   * // then you can get a profile:
+   * std::vector<float> x;
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: float->double everywhere?

File: src/DGtal/math/Profile.h [hunk 29]
```diff
@@ -0,0 +1,384 @@
...
+  public:
+
+    /**
+     * Used to specify the method to compute the profile values
+     * (used in @ref getProfile())
+     * 
+     **/
+    enum ProfileType{MEAN, MAX, MIN, MEDIAN};
+    
+    typedef TValueFunctor Functor;
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: doxygen doc for internal types missing

File: src/DGtal/math/Profile.ih [hunk 30]
```diff
@@ -0,0 +1,276 @@
...
+DGtal::Profile<TValueFunctor, TValue>::stopStatsSaving()
+{
+  for ( unsigned int i = 0; i < myStats->size(); ++i ){
+    ((*myStats).at( i )).terminate();
+  } 
+}
+
+template<typename TValueFunctor, typename TValue>
+inline
+DGtal::Profile<TValueFunctor, TValue>::Profile( const Profile & other ){
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.ih]: check indentation style

File: src/DGtal/math/Profile.h [hunk 31]
```diff
@@ -0,0 +1,384 @@
...
+  /////////////////////////////////////////////////////////////////////////////
+  // class Profile
+  /**
+   * Description of class 'Profile' <p> \brief Aim: This class can be
+   * used to represent a profile (PX, PY) defined from an input set of
+   * samples (Xi, Yi). From all the samples (Xk, Yk) having the same
+   * value Xk, the associated value PY is computed (by default) by the
+   * mean of the values Yk. Note that other definitions can be used
+   * (MAX, MIN or MEDIAN).
+   *
```
Comment (dcoeurjo) [inline src/DGtal/math/Profile.h]: maybe you should say that internally, each sample abscissa is an instance of DGtal::Statistic

File: ChangeLog.md [hunk 32]
```diff
@@ -18,6 +18,14 @@
    hull computation on both side of a point sequence.
    (Bertrand Kerautret, [#1073](https://github.com/DGtal-team/DGtal/pull/1073))
 
+ - LogScaleProfile: new class to represent a (multi)scale profile e.g. a sequence
+   of statistics on digital lengths parameterized by a scale.
```
Comment (dcoeurjo) [inline ChangeLog.md]: scale -> grid resolution for the last one ?

**Comment by dcoeurjo** [comment]
Thanks for the PR. Just minor comments.

File: src/DGtal/math/MeaningfulScaleAnalysis.h (hunk 16 above)
Comment (kerautret) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: yes thanks

File: src/DGtal/math/MeaningfulScaleAnalysis.h (hunk 18 above)
Comment (kerautret) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: better ;à

File: src/DGtal/math/MeaningfulScaleAnalysis.h (hunk 19 above)
Comment (kerautret) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: agree

File: src/DGtal/math/MeaningfulScaleAnalysis.h (hunk 21 above)
Comment (kerautret) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: yes thanks you are right

File: src/DGtal/math/MeaningfulScaleAnalysis.h (hunk 22 above)
Comment (kerautret) [inline src/DGtal/math/MeaningfulScaleAnalysis.h]: yes oups even if it was in doc ;)

File: src/DGtal/math/MeaningfulScaleAnalysis.ih (hunk 23 above)
Comment (kerautret) [inline src/DGtal/math/MeaningfulScaleAnalysis.ih]: yes I forget

File: src/DGtal/math/Profile.h (hunk 25 above)
Comment (kerautret) [inline src/DGtal/math/Profile.h]: yes

File: src/DGtal/math/Profile.h (hunk 26 above)
Comment (kerautret) [inline src/DGtal/math/Profile.h]: better yes

File: src/DGtal/math/Profile.h (hunk 28 above)
Comment (kerautret) [inline src/DGtal/math/Profile.h]: agree done everywhere

File: src/DGtal/math/Profile.h (hunk 31 above)
Comment (kerautret) [inline src/DGtal/math/Profile.h]: agree thanks

File: ChangeLog.md (hunk 32 above)
Comment (kerautret) [inline ChangeLog.md]: yes, even if with the thickness notion of Meaningful thickness it is not directly a grid size but almost equivalent

**Comment by kerautret** [comment]
@dcoeurjo done thanks for the review!

**Comment by dcoeurjo** [comment]
Thanks. Merging.

**Comment by kerautret** [comment]
Hello David,
Je crois que t’as fermé sans mergé ;)
A plus tard,
Bertrand

**Comment by dcoeurjo** [comment]
oops

**Pull request merged**